api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java

```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
         assertAttachSucceed(run(TaskStatus.RESUMING));
     }
 
+    @Test
+    public void shouldResolveProjectFiltering() {
+        final Folder project = new Folder();
+        project.setId(1L);
+
+        final Folder child = new Folder();
+        project.setId(2L);
+        project.getChildFolders().add(child);
+
+        final Pipeline pipeline1 = new Pipeline();
+        pipeline1.setId(2L);
+        project.getPipelines().add(pipeline1);
+
+        final Pipeline pipeline2 = new Pipeline();
+        pipeline2.setId(3L);
+        child.getPipelines().add(pipeline2);
+
+        final RunConfiguration configuration1 = new RunConfiguration();
+        configuration1.setId(4L);
+        project.getConfigurations().add(configuration1);
+
+        final RunConfiguration configuration2 = new RunConfiguration();
+        configuration2.setId(5L);
+        child.getConfigurations().add(configuration2);
+
+        when(folderApiService.load(project.getId())).thenReturn(project);
+
+        final PagingRunFilterVO filterVO = new PagingRunFilterVO();
+        filterVO.setProjectIds(Collections.singletonList(project.getId()));
+
+        final PipelineRunFilterVO.ProjectFilter projectFilter = pipelineRunManager.resolveProjectFiltering(filterVO);
+        assertEquals(2, projectFilter.getPipelineIds().size());
+        assertEquals(2, projectFilter.getConfigurationIds().size());
+    }
+
+    @Test
+    public void shouldThrowExceptionOnInexistentRunTagUpdate() {
+        doReturn(null).when(pipelineRunDao).loadPipelineRun(eq(NOT_EXISTING_RUN_ID));
+
+        final Runnable result = () -> pipelineRunManager.updateTags(NOT_EXISTING_RUN_ID, new TagsVO(null));
+        assertThrows(IllegalArgumentException.class, result);
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyCollections() {
+        final List<PipelineRunParameter> actualParameters =
+                pipelineRunManager.replaceParametersWithEnvVars(new ArrayList<>(), new HashMap<>());
+        assertTrue(CollectionUtils.isEmpty(actualParameters));
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyEnvVars() {
+        final List<PipelineRunParameter> parameters = new ArrayList<>();
+        parameters.add(new PipelineRunParameter(PARAM_NAME_1, TEST_STRING));
```

Comment:
Use `Collections.singletonList` for such list initialization.

Reply:
Done!